File: app/src/main/java/com/oriente/aptsample/startup/Dispatcher.java
package com.oriente.aptsample.startup;

import java.util.concurrent.Executor;

public interface Dispatcher {

    /**
     * 是否在主线程中执行
     */
    boolean callCreateOnMainThread();

    /**
     * 是否需要等待訪任务执行完成
     *
     * @return
     */
    boolean waitOnMainThread();

    /**
     * 等待
     */
    void toWait();

    /**
     * 有父任务执行完成
     * 计算器-1
     */
    void toNotify();

    Executor executor();

    /**
     * 线程优先级
     */
    int getThreadPriority();

}

File: startup/src/main/java/com/alex/startup/Dispatcher.kt
package com.alex.startup

import java.util.concurrent.Executor

interface Dispatcher {

    /**
     * 是否在主线程创建任务
     */
    fun callCreateOnMainThread()

    /**
     * 是否要等待该任务执行完成
     */
    fun waitOnMainThread(): Boolean

    /**
     * 利用CountDownLatch等待
     */
    fun toWait()

    fun toNotify()

    fun executor(): Executor

    /**
     * 线程优先级
     */
    fun getThreadPriority(): Int
}